src/Interpreters/MySQL/InterpretersMySQLDDLQuery.cpp

```diff
@@ -117,6 +118,37 @@ static inline NamesAndTypesList getColumnsList(ASTExpressionList * columns_defin
     return columns_name_and_type;
 }
 
+static ColumnsDescription getColumnsDescription(const NamesAndTypesList & columns_name_and_type, const ASTExpressionList * columns_definition)
+{
+    if (columns_name_and_type.size() != columns_definition->children.size())
+            throw Exception("Columns of different size provided.", ErrorCodes::BAD_ARGUMENTS);
+
+    ColumnsDescription columns_description;
+    ColumnDescription column_description;
+
+    for (
+        auto [column_name_and_type, declare_column_ast] = std::tuple{columns_name_and_type.begin(), columns_definition->children.begin()};
+        column_name_and_type != columns_name_and_type.end();
+        column_name_and_type++,
+        declare_column_ast++
```

Comment:
Seems a little over-complicated given that `column_name_and_type` was actually initialized as `getColumnsList(columns_definition)` a few line before call to this `getColumnsDescription`.

It could be simpler to first create `columns_description` (by using implementation of `getColumnsList` plus one check for `comment` in `column_options` - there are already such checks for other options so it will be a more correct place for that, and returning columns_description there instead) and then get `columns_name_and_type` from `columns_description->getAllPhysical()`. 

But I think it is also ok to leave your variant if you really want..

Reply:
hm, why there is a problem with `getKeys` - I was suggesting to only modify `getColumnsList` to return `ColumnsDescription`, instead of `NameAndTypesList`. And on the next line to get `columns_with_type_and_name` via `columns_description->getAllPhysical()`. All other methods will still use `columns_with_type_and_name`. This will simplify the code a lot and there will be no need for `getColumnsDescriptionFromList` method, which actually does the same thing, which is done in `getColumnsList` but for other column features.

Reply:
Since `getKeys `changes `columns_with_type_and_name` that we generate with `getColumnsList`, and if we just modify `getColumnsList` to return `ColumnsDescription` - we will miss this changes in our descriptions.

Reply:
Ok, I agree.
By the way, lets rename `getColumnsDescriptionFromList` -> `createColumnsDescription`